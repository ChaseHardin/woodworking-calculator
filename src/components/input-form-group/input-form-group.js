import React from 'react';
import Form from 'react-bootstrap/Form';

export const InputFormGroup = ({label, placeholder}) => {
    return (
        <Form.Group>
            {label && <Form.Label data-testid={label}>{label}</Form.Label>}
            <Form.Control placeholder={placeholder}></Form.Control>
        </Form.Group>
    );
}