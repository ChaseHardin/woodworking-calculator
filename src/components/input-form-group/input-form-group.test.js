import React from 'react';
import { render } from '@testing-library/react';
import { InputFormGroup } from './input-form-group'
import chance from '../../../test-utils/chance-wrapper';
import renderer from 'react-test-renderer';

let props;

beforeEach(() => {
    props = {
        label: chance.string(),
        placeholder: chance.string()
    };
});

test('renders form group with label and input', () => {
    const component = render(<InputFormGroup {...props} />);

    expect(component.getByTestId(props.label)).toHaveTextContent(props.label);
    expect(component.getByPlaceholderText(props.placeholder))
});

test('renders form group with input', () => {
    const component = render(<InputFormGroup placeholder={props.placeholder} />);

    expect(component.queryByTestId(props.label)).toBeNull();
    expect(component.getByPlaceholderText(props.placeholder))
});

test('renders snapshot of form group', () => {
    const tree = renderer.create(<InputFormGroup />).toJSON();

    expect(tree).toMatchSnapshot();
});