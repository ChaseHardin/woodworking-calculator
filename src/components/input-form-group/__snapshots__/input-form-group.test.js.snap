// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot of form group 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control"
  />
</div>
`;
